eppic-cli/src/main/java/eppic/assembly/Assembly.java

```diff
@@ -955,21 +974,64 @@ public String toString() {
 	}
 
 	/**
-	 * Assign a call to this Assembly, retrieve the call subsequently with {@link #getCall()}
+	 * Compute an initial (non-normalized) probabilistic score from
+	 * the individual interface probabilities.
 	 */
-	public void score() {
-
-		// we won't support non-fully-covering stoichiometries for the moment
-		// TODO support them: most likely requires a more complex data model where we can have score/calls for each of the subcomponents of the assembly
-		if (!assemblyGraph.isFullyCovering()) {
-			logger.warn("Assembly {} does not cover all entities, assembly scoring will be done for first sub-assembly only", toString());
-		}
+	public void calcScore() {
 		
+		// Construct a list of all the smaller powersets of this assembly that are equivalent
+		List<PowerSet> pss = new ArrayList<PowerSet>();
 		
-		SubAssembly firstSubAssembly = assemblyGraph.getSubAssemblies().get(0);
+		for (PowerSet ps : engagedSet.getOnPowerSet()) {
+			// Equivalent means that they have the same number of subassemblies
+			Assembly pa = new Assembly(crystalAssemblies, ps);
+			if (pa.getAssemblyGraph().getSubAssemblies().size() == 
+					this.getAssemblyGraph().getSubAssemblies().size())
+				pss.add(ps);
+		}
```

Comment:
Thanks for the explanation! it's clear now. That's a nice solution to the problem